### PR TITLE
chore: add headless interface comments and remove unused params

### PR DESCRIPTION
**Pull request by edwardfoyle**
The schema.json files are autogenerated. You only need to look at the .ts files for the diff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-headless-interface/schemas/api/1/AddApiRequest.schema.json [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+            "description": "The schema version.",
             "type": "number",
             "enum": [
                 1
             ]
         },
         "serviceConfiguration": {
-            "$ref": "#/definitions/AppSyncServiceConfiguration"
+            "$ref": "#/definitions/AppSyncServiceConfiguration",
+            "description": "The service configuration that will be interpretedy by Amplify."
```
Comment (jhockett) [inline packages/amplify-headless-interface/schemas/api/1/AddApiRequest.schema.json:14]: nit: typo "interpretedy"

Comment (edwardfoyle) [reply packages/amplify-headless-interface/schemas/api/1/AddApiRequest.schema.json]: lol good catch. fixed

File: packages/amplify-headless-interface/schemas/auth/1/AddAuthRequest.schema.json [hunk 2]
```diff
@@ -126,12 +138,15 @@
                     ]
                 },
                 "passwordPolicy": {
-                    "$ref": "#/definitions/CognitoPasswordPolicy"
+                    "$ref": "#/definitions/CognitoPasswordPolicy",
+                    "description": "If defined, specifies password constraint configuration. Defaut is minimum length of 8 characters."
```
Comment (jhockett) [inline packages/amplify-headless-interface/schemas/auth/1/AddAuthRequest.schema.json:142]: nit: typo "Defaut"

File: packages/amplify-headless-interface/src/interface/auth/add.ts [hunk 3]
```diff
@@ -1,147 +1,273 @@
...
+  /**
+   * If defined, specifies MFA configuration. Default is MFA off.
+   */
   mfa?: CognitoMFAConfiguration;
+  /**
+   * If defined, specifies password recovery configiuration. Default is email recovery.
+   */
   passwordRecovery?: CognitoPasswordRecoveryConfiguration;
+  /**
+   * If defined, specifies password constraint configuration. Defaut is minimum length of 8 characters.
```
Comment (jhockett) [inline packages/amplify-headless-interface/src/interface/auth/add.ts:121]: nit: "Defaut"

File: packages/amplify-headless-interface/schemas/auth/1/UpdateAuthRequest.schema.json [hunk 4]
```diff
@@ -90,9 +95,11 @@
                     ]
                 },
                 "passwordPolicy": {
-                    "$ref": "#/definitions/CognitoPasswordPolicy"
+                    "$ref": "#/definitions/CognitoPasswordPolicy",
+                    "description": "If defined, specifies password constraint configuration. Defaut is minimum length of 8 characters."
```
Comment (jhockett) [inline packages/amplify-headless-interface/schemas/auth/1/UpdateAuthRequest.schema.json:99]: nit: "Defaut"

File: packages/amplify-headless-interface/src/interface/auth/add.ts [hunk 5]
```diff
@@ -1,147 +1,273 @@
...
-    googleRecaptchaSecret: string;
-  };
 }
 
+/**
+ * Configuration for the AdminQueries API
+ */
 export interface CognitoAdminQueries {
+  /**
+   * Defines the API permissions. groupName must only be specified if restrictAccess is true, in whcih case only the specified user pool group will have access to the Admin Queries API.
```
Comment (jhockett) [inline packages/amplify-headless-interface/src/interface/auth/add.ts:246]: nit: "whcih"

**Review by jhockett — APPROVED** [review]
Found a few more typos, but otherwise LGTM

**Pull request merged**